xbmc/cores/AudioEngine/Engines/ActiveAE/AudioDSPAddons/ActiveAEDSP.cpp

```diff
@@ -148,17 +143,28 @@ void CActiveAEDSP::TriggerModeUpdate(bool bAsync /* = true */)
     m_databaseDSP.GetModes(m_modes[i], i);
   }
 
-  /*
-   * if any dsp processing is active restart playback
-   */
+  if (m_addonToDestroy.size() > 0)
+  {
+    for (std::list<AE_DSP_ADDON>::iterator iter = m_addonToDestroy.begin(); iter != m_addonToDestroy.end(); ++iter)
```

Comment:
No this is not enough. AudioDSP is a long lifetime object and it's add-ons can be enabled or disabled all the time. This will be required for binary add-on deployment. :-)